### DISCUSSION TITLE
Lifespan usage concurrently with @app.startup, @app.shutdown events

### DISCUSSION THREAD

**Discussion by stratosgear** [encode/starlette#2192] [Potential Issue] [answered]
As a Fastapi user, i found it very confusing that using the newly proposed lifespan methodology, SILENTLY ignores any @app.startup and/or @app.shutdown methods.

Fastapi does not mention anything like that in their documentation (although, for completeness sake, I DO believe that it should), and I cannot find anything on Starlette docs either.

There is a discussion on Fastapi (https://github.com/tiangolo/fastapi/discussions/9604) redirecting users to Starlette for the proper docs.

I am not claiming it is Starlette's fault, but without ANY kind of mentions or warnings, this is very tricky to debug.

Thanks!

**Comment by adriangb** [comment, marked as answer]
There should probably be a warning. If there's not, you're welcome to make a PR.

**Reply by Kludex** [reply]
We need to differentiate what comes from `self.router = Router()` on Starlette itself, and from `router = Router()`.

Some kind of marker or wrapper around the one passed through Starlette, I guess.

**Reply by stratosgear** [reply]
Not sure if it is good enough, but I just opened https://github.com/encode/starlette/pull/2193
